src/engine/resolvers.rs

```diff
@@ -80,11 +82,60 @@ where
         }
     }
 
+    /// Returns a neo4j database driver from the pool.
+    ///
+    /// # Errors
+    ///
+    /// Returns an [`Error`] variant [`DatabaseMismatch`] if the feature `neo4j`
+    /// is not enabled.
+    ///
+    /// Returns an [`Error]` variant [`Neo4jPoolGetConnectionFailed`] if a new
+    /// connection to the database cannot be retrieved from the pool.
+    ///
+    /// [`Error`]: ../../error/enum.Error.html
+    /// [`DatabaseMismatch`]: ../../error/enum.Error.html#variant.DatabaseMismatch
+    /// [`Neo4jPoolGetConnectionFailed`]: ../../error/enum.Error.html#variant.Neo4jPoolGetConnectionFailed
+    ///
+    /// # Examples
+    ///
+    /// ```rust, no_run
+    /// # use tokio::runtime::Runtime;
+    /// # use warpgrapher::engine::resolvers::{ResolverFacade, ExecutionResult};
+    ///
+    /// fn custom_resolve(facade: ResolverFacade<(), ()>) -> ExecutionResult {
+    ///     let mut rt = Runtime::new().unwrap();
+    ///     rt.block_on(async {
+    ///
+    ///         let neo4j_client = facade.db_as_neo4j().await.unwrap();
+    ///         // use neo4j client
+    ///
+    ///     });
+    ///     facade.resolve_null()
+    /// }
+    /// ```
+    #[cfg(feature = "neo4j")]
+    pub async fn db_as_neo4j(
+        &self,
+    ) -> Result<bb8::PooledConnection<'_, bb8_bolt::BoltConnectionManager>, Error> {
+        let db_pool: &bb8::Pool<bb8_bolt::BoltConnectionManager> =
+            match self.executor().context().pool() {
+                DatabasePool::Neo4j(p) => p,
+                _ => {
+                    return Err(Error::DatabaseMismatch {});
```

Comment:
Do we need a new error, or can we make Error::DatabaseNotFound broad enough to cover this case. It's already everywhere else that we look look for a database of a particular kind and can't find one.  If we really do need this to be a separate error, it should follow the NounVerb naming pattern of the rest of the errors, not NounNoun.

Reply:
If you are ok with it, I'm good to reuse "DatabaseNotFound".

Reply:
I'm good with reusing it.  The function caller is asking for a Neo4J database client.  A Neo4J database client is not found.

Reply:
Fixed.